error.go

```diff
@@ -0,0 +1,921 @@
+	
+	if len(e.metadata) > 0 {
+		sb.WriteString("\n  metadata:\n")
+		
+		for _, k := range slices.Sorted(maps.Keys(e.metadata)) {
+			v := e.metadata[k]
+			fmt.Fprintf(sb, "    - %s: %s visibility=%s\n", k, v.value, v.visibility.String())
+		}
+	}
+	
+	if e.id != nil {
+		fmt.Fprintf(sb, "\n  id: %s", *e.id)
+	}
+
+	if e.time != nil {
+		fmt.Fprintf(sb, "\n  time: %s", e.time.Format(time.RFC3339))
+	}
+
+	if e.subject != nil {
+		fmt.Fprintf(sb, "\n  subject: %s", *e.subject)
+	}
+
+	if e.sourceID != nil {
+		fmt.Fprintf(sb, "\n  source_id: %s", *e.sourceID)
+	}
+
+	if e.retryInfo != nil {
+		var retryStr string
+		if e.retryInfo.retryOffset != nil {
+			retryStr = fmt.Sprintf("retry_offset=%s", e.retryInfo.retryOffset.String())
+		} else if e.retryInfo.retryTime != nil {
+			retryStr = fmt.Sprintf("retry_time=%s", e.retryInfo.retryTime.Format(time.RFC3339))
+		}
+		
+		fmt.Fprintf(sb, "\n  retry_info: %s", retryStr)
+	}
+	
+
+	if e.help != nil && len(e.help.links) > 0 {
+		for i, link := range e.help.links {
+			if i > 0 {
+				sb.WriteString("\n")
+			}
+			fmt.Fprintf(sb, "- %s: %s", link.description, link.url)
+		}
+		sb.WriteString("\n")
+	}
+
+	if e.debugInfo != nil {
+		if e.debugInfo.detail != "" {
+			sb.WriteString(e.debugInfo.detail)
+		}
+			
+		for _, entry := range e.debugInfo.stackEntries {
+			sb.WriteString(entry)
+		}
+	}
+
+	return sb.String()
+}
+
+// Is implements error identity comparison for errors.Is
+func (e TrogonError) Is(target error) bool {
+	switch t := target.(type) {
+	case *TrogonError:
+		return e.domain == t.domain && e.reason == t.reason
+	case TrogonError:
+		return e.domain == t.domain && e.reason == t.reason
+	default:
+		return false
+	}
+}
+
+// Message returns the default message for a code
+func (c Code) Message() string {
+	// Since multiple codes can have the same HTTP value, we need to handle them explicitly
+	switch {
+	case c == Cancelled:
+		return "the operation was cancelled"
+	case c == Unknown:
+		return "unknown error"
+	case c == InvalidArgument:
+		return "invalid argument provided"
+	case c == DeadlineExceeded:
+		return "deadline exceeded"
+	case c == NotFound:
+		return "resource not found"
+	case c == AlreadyExists:
+		return "resource already exists"
+	case c == PermissionDenied:
+		return "permission denied"
+	case c == Unauthenticated:
+		return "unauthenticated"
+	case c == ResourceExhausted:
+		return "resource exhausted"
+	case c == FailedPrecondition:
+		return "failed precondition"
+	case c == Aborted:
+		return "operation aborted"
+	case c == OutOfRange:
+		return "out of range"
+	case c == Unimplemented:
+		return "not implemented"
+	case c == Internal:
+		return "internal error"
+	case c == Unavailable:
+		return "service unavailable"
+	case c == DataLoss:
+		return "data loss or corruption"
+	default:
+		return "unknown error"
+	}
+}
+
+// ToInt returns the HTTP status code for the error code
+func (c Code) ToInt() int {
+	switch c {
+	case Cancelled:
+		return 499
```

Comment:
why not use that as the value of the constants and avoid this entirely?

Reply:
because they are multiple codes that point to `500` http status, maybe I should change this name from `ToInt` to `ToHttpStatus`?

Reply:
I misunderstood your comment, what do you mean value as the constants?

Reply:
I renamed the function

Reply:
ah I missed that you have the same http code reused for multiple Code

if it was a 1:1 mapping I would just use the http code as value but you have multiple 500